testsuite/tools/expect_test.ml

```diff
@@ -185,12 +185,28 @@ let exec_phrase ppf phrase =
   if !Clflags.dump_source    then Pprintast.top_phrase ppf phrase;
   Toploop.execute_phrase true ppf phrase
 
+let ppx_rewrite = function
```

Comment:
Would it not be suitable to reuse `Toploop.preprocess_phrase` for this? (Isn't `"ocaml"` a reasonable choice of tool-name, assuming that some ppx producers may have special logic for it?) `preprocess_phrase` also checks `Clflags.dump_{parsetree,source}`, which sounds like something that could be handy in an except-test.

Reply:
(Toploop doesn't call `Ast_invariants`, but presumably this is an omission to be fixed?)

Reply:
Using `Toploop.preprocess_phase` sounds like the right choice, once the missing call to `Ast_invariants` is fixed. I will open a PR for those changes.